scripts/python/lib/utilities.py

```diff
@@ -17,4 +17,103 @@
 from __future__ import nested_scopes, generators, division, absolute_import, \
     with_statement, print_function, unicode_literals
 
+import os
+import re
+import subprocess
+import fileinput
+from shutil import copy2
+
+import lib.logger as logger
+
 PATTERN_MAC = '[\da-fA-F]{2}:){5}[\da-fA-F]{2}'
+
+
+def bash_cmd(cmd):
+    """Run command in Bash subprocess
+
+    Args:
+        cmd (str): Command to run
+    """
+    log = logger.getlogger()
+    command = ['bash', '-c', cmd]
+    log.debug('Run subprocess: %s' % ' '.join(command))
+    output = subprocess.check_output(command, universal_newlines=True)
+    log.debug(output)
+
+
+def backup_file(path):
+    """Save backup copy of file
+
+    Backup copy is saved as the name of the original with '.orig'
+    appended. The backup copy filemode is set to read-only.
+
+    Args:
+        path (str): Path of file to backup
+    """
+    log = logger.getlogger()
+    backup_path = path + '.orig'
+    log.debug('Make backup copy of orignal file: \'%s\'' % backup_path)
+    copy2(path, backup_path)
+    os.chmod(backup_path, 0o444)
+
+
+def append_line(path, line):
+    """Append line to end of text file
+
+    Args:
+        path (str): Path of file
+        line (str): String to append
+    """
+    log = logger.getlogger()
+    log.debug('Add line \'%s\' to file \'%s\'' % (line, path))
+    if not line.endswith('\n'):
+        line += '\n'
+    with open(path, 'a') as file_out:
+        file_out.write(line)
```

Comment:
close?

Reply:
Using `with` handles the file close automatically.